src/v3.rs

```diff
@@ -126,7 +133,22 @@ impl Sender {
             HeaderValue::from_static("application/json"),
         );
         headers.insert(header::USER_AGENT, HeaderValue::from_static("sendgrid-rs"));
+        Ok(headers)
+    }
 
+    #[cfg(feature = "async")]
+    pub fn send(&self, mail: &Message) -> impl Future<Item=Response, Error=SendgridError> {
+        let client = Client::new();
+        let headers = self.get_headers().expect("Error while set api key in header.");
```

Comment:
I'd prefer not to have an `expect` call here. Can we see about making this return a `Result` or having `get_headers` return a `HeaderMap` instead of a `Result`?

Reply:
I changed the implementation in send function. I transformed the Result into Future to pass it into the returned Future.